src/nvim/eval.c

```diff
@@ -4023,18 +4023,20 @@ eval6 (
         rettv->v_type = VAR_FLOAT;
         rettv->vval.v_float = f1;
       } else {
-        if (op == '*')
+        if (op == '*') {
           n1 = n1 * n2;
-        else if (op == '/') {
-          if (n2 == 0) {                /* give an error message? */
-            if (n1 == 0)
-              n1 = -0x7fffffffL - 1L;                   /* similar to NaN */
-            else if (n1 < 0)
-              n1 = -0x7fffffffL;
-            else
-              n1 = 0x7fffffffL;
-          } else
+        } else if (op == '/') {
+          if (n2 == 0) {                // give an error message?
+            if (n1 == 0) {
+              n1 = VARNUMBER_MIN;  // similar to NaN
```

Comment:
Note that Vim's help doesn't reflect reality, or I'm misunderstanding it.  Using 8.0.0600:

```
When 64-bit Number support is enabled:
	  0 / 0  = -0x8000000000000000	(like NaN for Float)
	 >0 / 0  =  0x7fffffffffffffff	(like positive infinity)
	 <0 / 0  = -0x7fffffffffffffff	(like negative infinity)
```

However:

```
:echo printf('%d %#x', 0/0, 0/0)
-9223372036854775808 0x8000000000000000
:echo printf('%d %#x', 1/0, 1/0)
9223372036854775807 0x7fffffffffffffff
:echo printf('%d %#x', -1/0, -1/0)
-9223372036854775807 0x8000000000000001
:echo printf('%d %d', 0/0, eval(0/0))
-9223372036854775808 -9223372036854775807
```

so we're at least consistent with Vim's inconsistency.